momepy/utils.py

```diff
@@ -71,24 +79,67 @@ def _generate_primal(graph, gdf_network, fields, multigraph, oneway_column=None)
             stacklevel=3,
         )
 
-    key = 0
-    for row in gdf_network.itertuples():
-        first = row.geometry.coords[0]
-        last = row.geometry.coords[-1]
+    if osmnx_like:
 
-        data = list(row)[1:]
-        attributes = dict(zip(fields, data, strict=True))
-        if multigraph:
-            graph.add_edge(first, last, key=key, **attributes)
-            key += 1
-
-            if oneway_column:
-                oneway = bool(getattr(row, oneway_column))
-                if not oneway:
-                    graph.add_edge(last, first, key=key, **attributes)
-                    key += 1
-        else:
-            graph.add_edge(first, last, **attributes)
+        gdf_network["geometry"] = gdf_network["geometry"].apply(
```

Comment:
If this will always be a GeoDatFrame, should we use the `geometry` attribute here?
```suggestion
        gdf_network.geometry = gdf_network.geometry.apply(
```